Purchases/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -324,21 +324,46 @@ class PurchasesOrchestrator {
     }
 
     @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    func purchase(sk2Product product: SK2Product,
-                  discount: StoreProductDiscountType?) async throws -> PurchaseResultData {
-        let result: Result<(CustomerInfo, Bool), Error> = await self.purchase(sk2Product: product, discount: discount)
+    func purchase(
+        sk2Product: SK2Product,
+        discount: StoreProductDiscountType?
+    ) async throws -> PurchaseResultData {
+        var options: Set<Product.PurchaseOption> = [
+            .simulatesAskToBuyInSandbox(Purchases.simulatesAskToBuyInSandbox)
+        ]
+
+        if let discount = discount {
+            let discount = try await self.promotionalOffer(
+                forProductDiscount: discount,
+                product: StoreProduct(sk2Product: sk2Product)
+            )
+            options.insert(discount.sk2PurchaseOption)
```

Comment:
perhaps we should log something here about adding a promo offer?

Reply:
Good idea.

Reply:
This actually belongs in #1217, I put it there.